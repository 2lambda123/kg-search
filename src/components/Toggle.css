/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

.kgs-toggle  {
    display: none;
    max-height: 40px;   
}

.kgs-toggle > div  {
    display: -ms-flexbox;
    display: flex;
    max-height: 40px;
    align-content: center;
    text-align: center;
    border-radius: 3px;
    white-space: nowrap;
}
/* 
.kgs-toggle > div:before {
    content: "View as";
    margin-top: 4px;
    margin-right: 20px;
    font-size: 22px;
    line-height: 30px;
    color: var(--color-15);
    white-space: nowrap;
} */

.kgs-toggle > div button {
    -ms-flex: 1 1;
    flex: 1 1;
    padding: 3px 35px; 
    background-color: var(--background-color-2);
    border: 1px solid var(--background-color-2);
    border-right: none;
    cursor: pointer;
    font-weight: 500;
    font-size: var(--font-size-small);
    line-height: 2em;
    color: #000;
    text-decoration: none;
    white-space: nowrap;
    -webkit-transition: -webkit-box-shadow 0.2s ease-in;
    transition: -webkit-box-shadow 0.2s ease-in;
    -o-transition: box-shadow 0.2s ease-in;
    transition: box-shadow 0.2s ease-in;
    transition: box-shadow 0.2s ease-in, -webkit-box-shadow 0.2s ease-in;
}

.kgs-toggle > div > button:first-child {
    border-radius: 3px 0 0 3px;
}

.kgs-toggle > div > button:last-child {
    border-radius: 0 3px 3px 0;
}

.kgs-toggle > div > button.is-active {
    background-color: var(--color-selected);
    border-color: var(--color-selected);
    cursor: default;
}

.kgs-toggle > div > button:not(.is-active):not(.is-disabled):hover,
.kgs-toggle > div > button:not(.is-active):not(.is-disabled):not([disabled]):focus  {
    z-index: 1;
}

.kgs-toggle > div > button:not(:last-child):not(.is-active):not(.is-disabled):hover,
.kgs-toggle > div > button:not(:last-child):not(.is-active):not(.is-disabled):not([disabled]):focus  {
    box-shadow: 0px 3px 6px var(--box-shadow-2);
}

.kgs-toggle > div > button:last-child:not(.is-active):not(.is-disabled):hover,
.kgs-toggle > div > button:last-child:not(.is-active):not(.is-disabled):not([disabled]):focus  {
    box-shadow: 3px 3px 6px var(--box-shadow-2);
}

.kgs-toggle > div > button > label {
    display: inline;
    font-size: 22px;
    font-weight: normal;
    color: var(--color-16);
}

.kgs-toggle > div > button:not(.is-active):hover > label {
    color: var(--color-11);
}

.kgs-toggle > div > button.is-active > label {
    color: white;
}
