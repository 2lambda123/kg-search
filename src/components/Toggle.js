/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

import React from "react";
import "./Toggle.css";

const ToggleItem = ({label, value, isActive, onClick}) => {
  const handleClick = () => {
    onClick(value);
  };
  return (
    <button type="button" className={(isActive?"is-active":"")}  disabled={isActive} onClick={handleClick} >
      <label>{label}</label>
    </button>
  );
};

export const Toggle = ({className, show, value, items, onClick}) => {
  if (!show) {
    return null;
  }
  const classNames = ["kgs-toggle", className].join(" ");
  return (
    <div className={classNames}>
      <div>
        {items.map(item =>
          <ToggleItem key={item.value} label={item.label} value={item.value} isActive={item.value === value} onClick={onClick} />
        )}
      </div>
    </div>
  );
};