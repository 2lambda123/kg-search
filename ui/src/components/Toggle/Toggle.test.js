/*
 * Copyright 2018 - 2021 Swiss Federal Institute of Technology Lausanne (EPFL)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This open source software code was developed in part or in whole in the
 * Human Brain Project, funded from the European Union's Horizon 2020
 * Framework Programme for Research and Innovation under
 * Specific Grant Agreements No. 720270, No. 785907, and No. 945539
 * (Human Brain Project SGA1, SGA2 and SGA3).
 *
 */

import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { mount, shallow, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Toggle from "./Toggle";

Enzyme.configure({ adapter: new Adapter() });

test('Toggle component renders initially', () => {
    const component = renderer.create(
        <Toggle className="className" show={true} value="a value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={() => {}} />
    );
  
    expect(component.toJSON()).toMatchSnapshot();
});

test('Toggle test show false"', () => {
    const component = renderer.create(
        <Toggle className="className" show={false} value="a value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={() => {}} />
    );
    expect(component.toJSON()).toBe(null);
});
  
test('Toggle test className"', () => {
    const component = shallow(
        <Toggle className="className" show={true} value="a value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={() => {}} />
    );
    expect(component.hasClass("className"));
});

test('Toggle test number of items', () => {
    const component = render(
        <Toggle className="className" show={true} value="a value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={() => {}} />
    );
    expect(component.find("button").length).toBe(2);
});

test('Toggle test item label', () => {
    const component = mount(
        <Toggle className="className" show={true} value="a value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={() => {}} />
    );
    expect(component.find("button").at(0).text()).toEqual("a label");
    expect(component.find("button").at(1).text()).toEqual("another label");
});

test('Toggle test active item', () => {
    const component = mount(
        <Toggle className="className" show={true} value="a value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={() => {}} />
    );
    expect(component.find("button").at(0).hasClass("is-active"));
    expect(component.find("button").at(1).hasClass("is-active")).toBe(false);
});

test('Toggle test active item', () => {
    const component = mount(
        <Toggle className="className" show={true} value="another value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={() => {}} />
    );
    expect(component.find("button").at(0).hasClass("is-active")).toBe(false);
    expect(component.find("button").at(1).hasClass("is-active"));
});

test('Toggle test no active items', () => {
    const component = mount(
        <Toggle className="className" show={true} value="a third value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={() => {}} />
    );
    expect(component.find("button").at(0).hasClass("is-active")).toBe(false);
    expect(component.find("button").at(1).hasClass("is-active")).toBe(false);
});

test('Toggle test  button click', () => {
    const fn = jest.fn();
    const component = mount(
        <Toggle className="className" show={true} value="a third value" items={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onClick={fn} />
    );
    component.find('button').at(1).simulate('click');
    expect(fn.mock.calls.length).toBe(1);
    expect(fn.mock.calls[0][0]).toBe("another value");
});