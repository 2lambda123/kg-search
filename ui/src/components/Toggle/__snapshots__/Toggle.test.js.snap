// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component renders initially 1`] = `
<div
  className="kgs-toggle className"
>
  <div>
    <button
      className="is-active"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <label>
        a label
      </label>
    </button>
    <button
      className=""
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <label>
        another label
      </label>
    </button>
  </div>
</div>
`;
